tests/countries/test_portugal.py

```diff
@@ -9,95 +9,352 @@
 #  Website: https://github.com/dr-prodigy/python-holidays
 #  License: MIT (see LICENSE file)
 
-import unittest
-from datetime import date
+from holidays.countries.portugal import Portugal, PT, PRT
+from tests.common import TestCase
 
-import holidays
 
+class TestPortugal(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(Portugal)
 
-class TestPT(unittest.TestCase):
     def setUp(self):
-        self.holidays = holidays.PT()
+        super().setUp()
+
+        years = range(1910, 2023)
```

Comment:
There is no need to init all these years if we're only going to use 1910 and 2017

Reply:
fixed.